save checkpoint error

when I run the SFT of Qwen3-30B-A3B on 2 nodes with 8 GPUs each, I meet the following error:
```
  File "Megatron-LM/megatron/core/optimizer/distrib_optimizer.py", line 1254, in sharded_state_dict
    param_state = self.sharded_param_state_dp_reshardable(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "Megatron-LM/megatron/core/optimizer/distrib_optimizer.py", line 1648, in sharded_param_state_dp_reshardable
    assert tensors[key].shape == (gbuf_local_end - gbuf_local_start,), (
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: (torch.Size([]), 0, 77856768)
```
What is the reason and how can I solve this.

--- a/docker/patch/latest/megatron.patch
+++ b/docker/patch/latest/megatron.patch
@@ -356,7 +356,7 @@ index a1230568c..1fd52f65a 100644
                      },
                  )
 diff --git a/megatron/core/optimizer/distrib_optimizer.py b/megatron/core/optimizer/distrib_optimizer.py
-index 6e093f96f..c1dfe205b 100644
+index 6e093f96f..eac21a3ea 100644
 --- a/megatron/core/optimizer/distrib_optimizer.py
 +++ b/megatron/core/optimizer/distrib_optimizer.py
 @@ -677,6 +677,8 @@ class DistributedOptimizer(MixedPrecisionOptimizer):
@@ -368,6 +368,15 @@ index 6e093f96f..c1dfe205b 100644
  
          # Grad scaler state.
          if self.grad_scaler:
+@@ -1646,6 +1648,8 @@ class DistributedOptimizer(MixedPrecisionOptimizer):
+                             if key == 'padding':
+                                 tensors[key] = LocalNonpersistentObject(tensors[key])
+                                 continue
++                            if key == 'step':
++                                continue
+                             assert tensors[key].shape == (gbuf_local_end - gbuf_local_start,), (
+                                 tensors[key].shape,
+                                 gbuf_local_start,
 diff --git a/megatron/core/parallel_state.py b/megatron/core/parallel_state.py
 index a273002b9..4f821cfd5 100644
 --- a/megatron/core/parallel_state.py

--- a/docker/version.txt
+++ b/docker/version.txt
@@ -1,1 +1,1 @@
-nightly-dev-20251222a
+nightly-dev-20251226b

